chromium/test/rules_test.js

```diff
@@ -96,14 +96,35 @@ describe('rules.js', function() {
   })
 
   describe('RuleSets', function() {
+    let rules_json = JSON.parse('[{"name":"Freerangekitten.com","rule":[{"to":"https:","from":"^http:"}],"target":["freerangekitten.com","www.freerangekitten.com"]}]');
+
+    beforeEach(function() {
+      this.rsets = new RuleSets();
+    });
+
+    describe('#addFromJson', function() {
+      it('can add a rule', function() {
+        this.rsets.addFromJson(rules_json);
+
+        assert.isTrue(this.rsets.targets.has('freerangekitten.com'));
+      });
+    });
+
+    describe('#rewriteURI', function() {
+      it('rewrites host added from json', function() {
+        let host = 'freerangekitten.com';
+        this.rsets.addFromJson(rules_json);
+
+        let newuri = this.rsets.rewriteURI('http://' + host + '/', host);
+
+        assert.strictEqual(newuri, 'https://' + host + '/', 'protocol changed to https')
```

Comment:
Could you also add a test for the no-match case, please? (it returns `null` as far as I remember)